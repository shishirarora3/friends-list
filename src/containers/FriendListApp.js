import React, {Component} from 'react';
import styles from './FriendListApp.css';
import {connect} from 'react-redux';
import size from "lodash/size";
import {addFriend, deleteFriend, starFriend, prevPage, nextPage, updateSex} from '../actions';
import {AddFriendInput, FriendList, Pagination} from '../components';
import {getFriendsByPage} from "../selectors/FriendsByPage";

export class FriendListApp extends Component {

  render() {
    const {
      friendlist: {friendsById}, addFriend, deleteFriend, starFriend, updateSex,
      nextPage, prevPage,
      page: {
        pageNo
      }
    } = this.props;
    const pageSize = 2;
    const totalFriends = size(friendsById);
    const friendActions = {
      addFriend,
      deleteFriend,
      starFriend,
      updateSex
    };
    const friendsByPage = getFriendsByPage({friendsById, pageNo, pageSize});
    const pageActions = {
      nextPage,
      prevPage
    };
    return (
      <div className={styles.friendListApp}>
        <h1>The FriendList</h1>
        <AddFriendInput addFriend={friendActions.addFriend}/>
        <FriendList friends={friendsByPage} actions={friendActions}/>
        <Pagination pageNo={pageNo} totalFriends={totalFriends}
                    pageSize={pageSize} actions={pageActions}/>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return state
}

export default connect(mapStateToProps, {
  addFriend,
  deleteFriend,
  starFriend,
  nextPage,
  prevPage,
  updateSex
})(FriendListApp)
