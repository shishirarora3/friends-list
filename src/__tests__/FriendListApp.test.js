import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {FriendListApp} from '../containers/FriendListApp';
import toJson from 'enzyme-to-json';

const thunk = ({ dispatch, getState }) => next => action => {
  if (typeof action === 'function') {
    return action(dispatch, getState)
  }
  return next(action);
};

const create = () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn()
  };
  const next = jest.fn();
  const invoke = action => thunk(store)(next)(action);
  return { store, next, invoke };
};

/*it('passes through non-function action', () => {
  const { next, invoke } = create();
  const action = { type: 'TEST' };
  invoke(action);
  expect(next).toHaveBeenCalledWith(action)
});*/
it('renders the component', () => {
  const tree = renderer.create(<FriendListApp />).toJSON();
  expect(tree).toMatchSnapshot();
});

