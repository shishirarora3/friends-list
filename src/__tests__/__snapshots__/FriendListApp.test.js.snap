exports[`test renders the component 1`] = `"<div>Hedlo Jest</div>"`;
